### DISCUSSION TITLE
Conditions do not limit base subject

### DISCUSSION THREAD

**Discussion by ronderksen** [stalniy/casl#985] [Q&A]
Let's say I have the following rule:
```
can("update", "post", {isOwner: true"})
```
Then I would expect `can("update", "post")` to return `false` (because the rule needs the condition). Is this an achievable result at all? I've tried adding `cannot("update", "post") before or after the `can` rule, but it doesn't help.

**Comment by stalniy** [comment]
This is explained in the guide  https://casl.js.org/v6/en/guide/intro#checking-logic

**Reply by stalniy** [reply]
To always check conditions you need to pass subject/entity not its type. There is no logical way to check conditions on entity type because it’s not an entity 

conditions are defined on entity but you do check on entity type. This is 2 different checks for permission system

**Reply by stalniy** [reply]
With typescript you can restrict the second parameter, not sure how it will work with AbilityBuilder though. You can restrict it to use only entities and forbid type based checking
